src/webgpu/shader/validation/wgsl/variable.spec.ts

```diff
@@ -0,0 +1,198 @@
+const kTypeConstructorExpression = [
+  { rhs_type: 'i32', rhs_value: '0' },
+  { rhs_type: 'f32', rhs_value: '0.0' },
+  { rhs_type: 'u32', rhs_value: '0u' },
+  { rhs_type: 'bool', rhs_value: 'false' },
+] as const;
+
+g.test('v_0033_scalar')
+  .params(params().combine(kTypeConstructorExpression).combine(poptions('lhs_type', kSclarType)))
+  .fn(async t => {
+    const { lhs_type, rhs_type, rhs_value } = t.params;
+    const code = `
+
+    [[stage(vertex)]]
+    fn main() -> void {
+      var v : ${lhs_type} = ${rhs_type}(${rhs_value});
+    }`;
+    if (lhs_type === rhs_type) {
+      t.expectCompileResult(true, code);
+    } else {
+      t.expectCompileResult('v-0033', code);
+    }
+  });
+
+g.test('v_0033_vec2_e1_e2')
+  .params(
+    params()
+      .combine(poptions('lhs_type', kSclarType))
+      .combine(poptions('rhs_type', kSclarType))
+      .combine(poptions('e1_type', kSclarType))
+      .combine(poptions('e2_type', kSclarType))
+  )
+  .fn(async t => {
+    const { lhs_type, rhs_type, e1_type, e2_type } = t.params;
+    const code = `
+
+    [[stage(vertex)]]
+    fn main() -> void {
+      var e1: ${e1_type} = ${e1_type}();
+      var e2: ${e2_type} = ${e2_type}();
+      var v : vec2<${lhs_type}> = vec2<${rhs_type}>(e1, e2);
+    }`;
+    if (lhs_type === rhs_type && rhs_type === e1_type && e1_type === e2_type) {
+      t.expectCompileResult(true, code);
+    } else {
+      t.expectCompileResult('v-0033', code);
+    }
+  });
+
+g.test('v_0033_vec2_e')
+  .params(
+    params().combine(poptions('lhs_type', kSclarType)).combine(poptions('rhs_type', kSclarType))
+  )
+  .fn(async t => {
+    const { lhs_type, rhs_type } = t.params;
+    const code = `
+
+    [[stage(vertex)]]
+    fn main() -> void {
+      var e : vec2<${rhs_type}> = vec2<${rhs_type}>();
+      var v : vec2<${lhs_type}> = e;
+    }`;
+    if (lhs_type === rhs_type) {
+      t.expectCompileResult(true, code);
+    } else {
+      t.expectCompileResult('v-0033', code);
+    }
+  });
+
+g.test('v_0033_vec3_e1_e2_e3')
+  .params(
+    params()
+      .combine(poptions('lhs_type', kSclarType))
+      .combine(poptions('rhs_type', kSclarType))
+      .combine(poptions('e1_type', kSclarType))
+      .combine(poptions('e2_type', kSclarType))
+      .combine(poptions('e3_type', kSclarType))
+  )
+  .fn(async t => {
+    const { lhs_type, rhs_type, e1_type, e2_type, e3_type } = t.params;
+    const code = `
+
+    [[stage(vertex)]]
+    fn main() -> void {
+      var e1: ${e1_type} = ${e1_type}();
+      var e2: ${e2_type} = ${e2_type}();
+      var e3: ${e3_type} = ${e3_type}();
+      var v : vec3<${lhs_type}> = vec3<${rhs_type}>(e1, e2, e3);
+    }`;
+    if (
+      lhs_type === rhs_type &&
+      rhs_type === e1_type &&
+      e1_type === e2_type &&
+      e2_type === e3_type
+    ) {
+      t.expectCompileResult(true, code);
+    } else {
+      t.expectCompileResult('v-0033', code);
+    }
+  });
+
+g.test('v_0033_vec3_e1_e2')
```

Comment:
@kainino0x 
Is it possible to combine g.test('v_0033_vec3_e1_e2') with g.test('v_0033_vec3_e2_e1')?